contributors/design-proposals/alternate-api-representations.md

```diff
@@ -0,0 +1,405 @@
+# Alternate representations of API resources
+
+## Abstract
+
+Naive clients benefit from allowing the server to returning resource information in a form
+that is easy to represent or is more efficient when dealing with resources in bulk. It
+should be possible to ask an API server to return a representation of one or more resources
+of the same type in a way useful for:
+
+* Retrieving a subset of object metadata in a list or watch of a resource, such as the
+  metadata needed by the generic Garbage Collector or the Namespace Lifecycle Controller
+* Dealing with generic operations like `Scale` correctly from a client across multiple API
+  groups, versions, or servers
+* Return a simple tabular representation of an object or list of objects for naive
```

Comment:
Maybe we could suggest clients which fields are mandatory and which contains extended data as a sort of annotation in the openapi spec (if possible), instead doing this? I'm worried this will cause us some headaches in the long run.